### ISSUE TITLE
[null-safety] False positive on JavaScript `final clazz =Type.getClass(...)`

### ISSUE THREAD

**Issue by sebthom** [HaxeFoundation/haxe#10275]
The following code works fine on all targets except on JavaScript where it throws `Null safety: Cannot assign nullable value here.`

https://try.haxe.org/#7f5CE35f

```haxe
@:nullSafety
class Test {
  static function main() {
    final clazz = Type.getClass(new Array()); // <-- fails to compile
  }
}
```

**Comment by NotHyper-474** [comment]
Looks like this was only resolved in Haxe 5, is there a chance the fix is going to be back-ported into 4.3 too?

**Comment by Simn** [comment]
No, some of the underlying architecture changed too much for that to be viable.
